src/utils/convert.js

```diff
@@ -1,798 +1,164 @@
-  trigger: 'triggers',
-  search: 'searches',
-  create: 'actions'
-};
-
-const stepTypeMap = {
-  trigger: 'triggers',
-  search: 'searches',
-  create: 'creates'
-};
-
-// map CLI step names to verbs for display labels
-const stepVerbsMap = {
-  trigger: 'Get',
-  create: 'Create',
-  search: 'Find'
-};
-
-// map CLI step names to templates for descriptions
-const stepDescriptionTemplateMap = {
-  trigger: _.template('Triggers on a new <%= lowerNoun %>.'),
-  create: _.template('Creates a <%= lowerNoun %>.'),
-  search: _.template('Finds a <%= lowerNoun %>.')
-};
 
-const renderTemplate = (templateFile, templateContext, prettify = true) => {
-  return readFile(templateFile)
-    .then(templateBuf => templateBuf.toString())
-    .then(template =>
-      _.template(template, { interpolate: /<%=([\s\S]+?)%>/g })(templateContext)
-    )
-    .then(content => {
-      if (prettify) {
-        const ext = path.extname(templateFile).toLowerCase();
-        const prettifier = {
-          '.json': origString =>
-            JSON.stringify(JSON.parse(origString), null, 2),
-          '.js': origString =>
-            prettier.format(origString, {
-              singleQuote: true,
-              printWidth: 120
-            })
-        }[ext];
-        if (prettifier) {
-          return prettifier(content);
-        }
-      }
-      return content;
-    });
-};
+// A placeholder that can be used to identify this is something we need to replace
+// before generating the final code. See replacePlaceholders function. Make it really
+// special and NO regex reserved chars.
+const REPLACE_DIRECTIVE = '__REPLACE_ME@';
 
-const createFile = (content, fileName, dir) => {
-  const destFile = path.join(dir, fileName);
+const makePlaceholder = replacement => `${REPLACE_DIRECTIVE}${replacement}`;
 
-  return ensureDir(path.dirname(destFile))
-    .then(() => writeFile(destFile, content))
-    .then(() => {
-      startSpinner(`Writing ${fileName}`);
-      endSpinner();
-    });
-};
-
-const renderProp = (key, value) => `${key}: ${value}`;
+const replacePlaceholders = str =>
+  str.replace(new RegExp(`"${REPLACE_DIRECTIVE}([^"]+)"`, 'g'), '$1');
 
 const quote = s => `'${s}'`;
 
 const escapeSpecialChars = s => s.replace(/\n/g, '\\n').replace(/'/g, "\\'");
 
-const getAuthType = definition => {
-  return authTypeMap[definition.general.auth_type];
-};
-
-const hasAuth = definition => {
-  return (
-    getAuthType(definition) !== 'custom' && !_.isEmpty(definition.auth_fields)
+const createFile = async (content, filename, dir) => {
+  const destFile = path.join(dir, filename);
+  await ensureDir(path.dirname(destFile));
+  await writeFile(destFile, content);
+  startSpinner(`Writing ${filename}`);
+  endSpinner();
+};
+
+const prettifyJs = code => prettier.format(code, { singleQuote: true });
+
+const renderTemplate = async (
+  templateFile,
+  templateContext,
+  prettify = true
+) => {
+  const templateBuf = await readFile(templateFile);
```

Comment:
I believe this can be `const template = await readFile(templateFile, 'utf-8')`, since passing encoding gives you a string instead of a buffer.

Reply:
This `readFile` is a function we wrote in `src/utils/files.js` and doesn't accept an encoding option. It is used in other places, so I rather not change its interface for now.